### PR TITLE
Add the capi image as dependency to baremetalds-assisted-setup

### PR DESCRIPTION
**Pull request by eranco74**

### CODE REVIEW

File: ci-operator/step-registry/baremetalds/assisted/setup/baremetalds-assisted-setup-ref.yaml [hunk 1]
```diff
@@ -19,6 +19,8 @@ ref:
     env: ASSISTED_AGENT_IMAGE
   - name: "assisted-installer-controller"
     env: ASSISTED_CONTROLLER_IMAGE
+  - name: "assisted-installer-controller"
+    env: PROVIDER_IMAGE
```
Comment (osherdp) [inline ci-operator/step-registry/baremetalds/assisted/setup/baremetalds-assisted-setup-ref.yaml:23]: It seems reasonable as long as we:
* start now adding it anywhere needed. No reason we wouldn't add the dependency to all our owned files. release-controllers and external repos can be handled separately 
* add a comment here explaining that this is deliberate and not a mistake

File: ci-operator/step-registry/baremetalds/assisted/setup/baremetalds-assisted-setup-commands.sh [hunk 2]
```diff
@@ -101,6 +101,13 @@ echo "export SERVICE=${ASSISTED_SERVICE_IMAGE}" >> /root/config
 echo "export AGENT_DOCKER_IMAGE=${ASSISTED_AGENT_IMAGE}" >> /root/config
 echo "export CONTROLLER_IMAGE=${ASSISTED_CONTROLLER_IMAGE}" >> /root/config
 echo "export INSTALLER_IMAGE=${ASSISTED_INSTALLER_IMAGE}" >> /root/config
+# Most jobs and tests don't require this image, so we allow as optional
+# This is required due to the nounset flag
+PROVIDER_IMAGE=${PROVIDER_IMAGE:-}
+if [ ! -z ${PROVIDER_IMAGE} ];
+then
+  echo "export PROVIDER_IMAGE=${PROVIDER_IMAGE}" >> /root/config
```
Comment (osherdp) [inline ci-operator/step-registry/baremetalds/assisted/setup/baremetalds-assisted-setup-commands.sh:109]: assuming we sometimes put here the wrong image,
can we check it before? something like:
```
image_name=$(podman image inspect ${PROVIDER_IMAGE} -f "{{.RepoTags}}")
if [[ ${image_name} ~= "cluster-api-provider-agent" ...
```

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/baremetalds/assisted/setup/baremetalds-assisted-setup-commands.sh (hunk 2 above)
Comment (eranco74) [reply ci-operator/step-registry/baremetalds/assisted/setup/baremetalds-assisted-setup-commands.sh:109]: I'll update the if to check if statement too check:
if [ "${PROVIDER_IMAGE}" != "${CONTROLLER_IMAGE}" ]

**Review by osherdp — APPROVED** [review]

**Pull request merged**